hack/test/trap

```diff
@@ -0,0 +1,106 @@
+#!/bin/bash
+#
+# This test ensures that the functions in 'os::util::trap' function as expected
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+OS_ROOT=$(dirname "${BASH_SOURCE}")/../..
+source "${OS_ROOT}/hack/cmd_util.sh"
+source "${OS_ROOT}/hack/lib/util/trap.sh"
+cd "${OS_ROOT}"
+
+# basic functionality tests
+some_funcname="func"
+VERBOSE=true os::cmd::expect_success "os::util::trap::add ${some_funcname} ERR"
+os::cmd::expect_success_and_text "echo ${REGISTERED_TRAPS_TO_SIGNALS[${some_funcname}]-}" "ERR"
```

Comment:
I've been hoisted by my own petard. `os::cmd` runs the command in a sub-shell and therefore won't be able to make any changes to the envars of the parent shell.